app/src/main/java/ch/epfl/sdp/appart/database/FirestoreDatabaseService.java

```diff
@@ -235,16 +235,16 @@ public FirestoreDatabaseService() {
         }
         return updateUserDb(isFinishedFuture, user);*/
         CompletableFuture<Boolean> result = new CompletableFuture<>();
-        result.complete(true);
-        return result;
+        return updateUserDb(result, user);
 
     }
 
     private CompletableFuture<Boolean> updateUserDb(CompletableFuture<Boolean> res, User user){
-            db.document(user.getUserId())
-            .set(extractUserInfo(user))
-            .addOnCompleteListener(
-                task -> res.complete(task.isSuccessful()));
+            db.collection(UserLayout.DIRECTORY)
+                    .document(user.getUserId())
+                    .set(extractUserInfo(user))
+                    .addOnCompleteListener(
+                            task -> res.complete(task.isSuccessful()));
```

Comment:
Add a failure case if task.isSuccessful is false

Reply:
Good point! **`updateUser`** in **FirestoreDatabaseService** returns `task.isSuccessful()` to `UserViewModel` and then `UserProfileActivity`. I added a test case in the UserProfileActivity which displays an error snake (similar to Login) if `task.isSuccessful()` was returned **false.**

DEMO
---------------------

https://user-images.githubusercontent.com/47753346/116056938-991f7d80-a67e-11eb-9193-c4ac357ad75d.mov

Reply:
I think it makes more sense to do a completeExceptionally, because on a thenAccept the caller thinks everything went fine. The goal is to have the caller have a thenAccept for a succesful future, and a exceptionally where it can recover from failure.

Reply:
The **updateUserDb(...)** method in FirestoreDatabaseService will be **completely** refactored in the next PR in order to update the profile picture together with the user information. I think it could be clever to address this remark in the next PR once the **updateUserDb(...)** function will be completed. Thank you for the remark!